editor/filesystem_dock.cpp

```diff
@@ -1599,6 +1599,15 @@ void FileSystemDock::_update_dependencies_after_move(const HashMap<String, Strin
 	}
 }
 
+void FileSystemDock::_update_folder_colors_after_move(const HashMap<String, String> &p_folders_renames) {
+	for (const KeyValue<String, String> &rename : p_folders_renames) {
+		if (assigned_folder_colors.has(rename.key)) {
+			assigned_folder_colors[rename.value] = assigned_folder_colors[rename.key];
+			assigned_folder_colors.erase(rename.key);
```

Comment:
Actually, while you don't need to do empty checks, the settings should be saved after this operation. (do check if a path was changed though)

Reply:
Not needed, that's taken care of by `_update_project_settings_after_move` which is called after in both cases

Reply:
Can alternatively remove `const` from `_update_project_settings_after_move` and move the code there

Reply:
If it's done automatically then it's fne.